compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/LocalDeclarationsLowering.kt

```diff
@@ -163,36 +165,55 @@ class LocalDeclarationsLowering(
         override lateinit var transformedDeclaration: IrConstructor
     }
 
-    private class LocalClassContext(val declaration: IrClass, val inInlineFunctionScope: Boolean) : LocalContext() {
+    private inner class LocalClassContext(
+        val declaration: IrClass,
+        val inInlineFunctionScope: Boolean,
+        val constructorContext: LocalContext?
+    ) : LocalContext() {
         lateinit var closure: Closure
 
         // NOTE: This map is iterated over in `rewriteClassMembers` and we're relying on
         // the deterministic iteration order that `mutableMapOf` provides.
-        val capturedValueToField: MutableMap<IrValueDeclaration, IrField> = mutableMapOf()
+        val capturedValueToField: MutableMap<IrValueDeclaration, Lazy<IrField>> = mutableMapOf()
```

Comment:
This is minor, but maybe make it a bit more explicit, by using a custom class instead of `Lazy`? It'll make it easier to inspect where exactly the related logic is (searching for usages of Lazy is costly). E.g. (not sure about the names)
```
    private class FieldToBeGenerated(generate: () -> IrField) {
        private var generate = lazy(LazyThreadSafetyMode.NONE, generate)

        fun get(): IrField = generate.value

        val isAccessed: Boolean get() = generate.isInitialized()
    }
```

Reply:
Done. Rather than passing a function it's probably better to only have a symbol, then create IrFields for the used ones later.